### DISCUSSION TITLE
Anyway to skip 200 request logs?

### DISCUSSION THREAD

**Discussion by saikoml** [pocketbase/pocketbase#3800] [Q&A]
Is there a way to log ONLY the failed requests with status 400 or higher?

I want to use retention policy of 30 days but storing everything for that long will take too much space.

**Comment by ganigeorgiev** [comment]
At the moment one slightly hacky workaround could be to attach directly to `logsDao` hook in the Go code and "skip" the persistence action for lower statuses. Something like:

```go
app.OnAfterBootstrap().Add(func(e *core.BootstrapEvent) error {
    app.LogsDao().BeforeCreateFunc = func(eventDao *daos.Dao, m models.Model, action func() error) error {
        requestModel, ok := m.(*models.Request);
        if ok && requestModel.Status < 400 {
            return nil // skip
        }
        return action()
    }
})
```

**But be aware that this will no longer be necessary in the next v0.20.0 release and will also no longer work because the `models.Request` will be removed!**

I haven't pushed it yet as there are still some unit tests I need to finalize, but with the upcoming v0.20.0 release all logs will be generalized and we will support the standard Go 1.21 [slog Handler interface](https://pkg.go.dev/log/slog#Handler).

This means that the activity logs will be just a call to `app.Logger().Info(...)` or `app.Logger().Error(...)` (_for the failed echo handlers_) call and you'll be able to control the minimum [log level](https://pkg.go.dev/log/slog#Level) from the Admin UI settings (by default it will be set to `0/info`).
Also note that as of now the PocketBase slog handler is implemented to write the logs with debounce and on batches (_aka. there is a minor tradeoff between performance and durability_):
- 3 seconds after the last debounced log write
- when the batch threshold, currently 200, is reached (this is primarily to prevent the memory to grow unconstrained)
- right before app termination to attempt saving whatever is in the existing logs queue
